File: scholar-web/src/main/webapp/app/model/administration/settings/subject/SubjectSearch.js
Ext.define('scholar.model.administration.settings.subject.SubjectSearch', {
	extend : 'Ext.data.Model',
	idProperty: 'id',
	fields : [{
		name : 'id',
		type : 'int'
	},  {
		name : 'subjectName',
		type : 'string'
	},
	{
		name : 'subjectCode',
		type : 'string'
	},
	{
		name : 'examinationRequired',
		type : 'bool'
	},
	{
		name : 'startDate',
		type : 'date',
		dateFormat : 'n/j h:ia'
	}, {
		name : 'endDate',
		type : 'date',
		dateFormat : 'n/j h:ia'
	}, {
		name : 'lastChange',
		type : 'date',
		dateFormat : 'n/j h:ia'
	} ]

});

File: scholar-business/src/main/java/com/varun/scholar/business/data/entities/administration/settings/SubjectSearch.java
package com.varun.scholar.business.data.entities.administration.settings;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.varun.scholar.business.data.entities.AbstractEntity;


@Entity
public class SubjectSearch extends AbstractEntity {
	String subjectName;
	String subjectCode;
	boolean examinationRequired;
	@Temporal(TemporalType.DATE)
	Date startDate;
	@Temporal(TemporalType.DATE)
	Date endDate;
	@Temporal(TemporalType.DATE)
	Date lastChange;
}
